components/Builder.tsx

```diff
@@ -169,25 +176,42 @@ export default function Builder({ data }: { data: any }) {
         />
       </div>
 
-      <div className="flex gap-2 justify-center items-center">
-        {isSignedIn && (
-          <SaveButton
-            formation={formation}
-            artifact={artifact}
-            layout={layout}
-            user={user}
+      <div className="flex gap-2 items-center justify-between sm:w-full w-[min(100%,360px)] px-6">
+        <div className="flex gap-2">
+          {isSignedIn && (
+            <SaveButton
+              artifact={artifact}
+              formation={formation}
+              layout={layout}
+              user={user}
+            />
+          )}
+          <Button onClick={onDownloadButtonClick} className="h-8 px-2">
+            <Download />
+          </Button>
+        </div>
+        <div className="flex gap-2 justify-end">
+          <Input
+            className="w-[min(150px,100%)]"
+            hasClearInput
+            onChange={(e) =>
+              setCharacterFilter((prev) => ({ ...prev, name: e.target.value }))
+            }
+            onClearClick={() => {
+              setCharacterFilter((prev) => ({ ...prev, name: "" }));
+              searchInputRef.current?.focus();
```

Comment:
So the ref is just to hold focus on the input?

Reply:
Yeah - when you clear the input to prevent an extra user click if they want to retype it auto focuses. I should test this on mobile though if it re-opens the keyboard that might not be great.

Reply:
okay makes sense. that is what google does when you click the `x` so i'm all for it.